dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -78,45 +67,67 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
             THROW_IF_FAILED(notificationPlatform->AddToastRegistrationMapping(processName.get(), toastAppId.c_str()));
         }
 
-        THROW_IF_FAILED(::CoRegisterClassObject(
-            AppModel::Identity::IsPackagedProcess() ? details.TaskClsid() : winrt::guid(storedComActivatorString),
-            winrt::make<AppNotificationActivationCallbackFactory>().get(),
-            CLSCTX_LOCAL_SERVER,
-            REGCLS_MULTIPLEUSE,
-            &s_notificationComActivatorRegistration));
+        {
+            auto lock{ m_registrationLock.lock_exclusive() };
+            THROW_HR_IF_MSG(E_INVALIDARG, m_notificationComActivatorRegistration, "Already Registered for App Notifications!");
+            THROW_IF_FAILED(::CoRegisterClassObject(
+                AppModel::Identity::IsPackagedProcess() ? details.TaskClsid() : winrt::guid(storedComActivatorString),
+                winrt::make<AppNotificationActivationCallbackFactory>().get(),
+                CLSCTX_LOCAL_SERVER,
+                REGCLS_MULTIPLEUSE,
+                &m_notificationComActivatorRegistration));
+        }
 
         GetWaitHandleForArgs().create();
     }
 
+    void AppNotificationManager::Register()
```

Comment:
Please don't create this, I am modifying the existing one in my PR. That should suffice. Thanks.

Reply:
Just rebase